Measurements/Measurements.ipynb

```diff
@@ -0,0 +1,571 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To begin, first prepare this notebook for execution (if you skip the first step, you'll get \"Syntax does not match any known patterns\" error when you try to execute Q# code in the next cells; if you skip the second step, you'll get \"Invalid kata name\" error):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%package Microsoft.Quantum.Katas"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%workspace reload"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part I. Discriminating Orthogonal States"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.1. $|0\\rangle$ or $|1\\rangle$?\n",
+    "\n",
+    "**Input:** A qubit which is guaranteed to be in $|0\\rangle$ or $|1\\rangle$ state.\n",
+    "\n",
+    "**Output:**  `true` if the qubit was in the $|1\\rangle$ state, or `false` if it was in the $|0\\rangle$ state. The state of the qubit at the end of the operation does not matter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T101_IsQubitOne_Test \n",
+    "\n",
+    "operation IsQubitOne (q : Qubit) : Bool {\n",
+    "    // The operation M will measure a qubit in the Z basis (|0⟩ and |1⟩ basis)\n",
+    "    // and return Zero if the observed state was |0⟩ or One if the state was |1⟩.\n",
+    "    // To answer the question, you need to perform the measurement and check whether the result\n",
+    "    // equals One - either directly or using library function IsResultOne.\n",
+    "    //\n",
+    "    // Type the following: return M(q) == One;\n",
+    "    // Then run the cell using Ctrl/⌘+Enter.\n",
+    "\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.2. Set the qubit to the $|0\\rangle$ state.\n",
+    "\n",
+    "**Input:** A qubit in an arbitrary state.\n",
+    "\n",
+    "**Goal:**  Change the state of the qubit to $|0\\rangle$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T102_InitializeQubit_Test \n",
+    "\n",
+    "operation InitializeQubit (q : Qubit) : Unit {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.3. $|+\\rangle$ or $|-\\rangle$?\n",
+    "\n",
+    "**Input:** A qubit which is guaranteed to be in $|+\\rangle$ or $|-\\rangle$ state. As a reminder, $|+\\rangle = \\frac{1}{\\sqrt{2}} \\big(|0\\rangle + |1\\rangle\\big)$, $|-\\rangle = \\frac{1}{\\sqrt{2}} \\big(|0\\rangle - |1\\rangle\\big)$.\n",
```

Comment:
A qubit which is guaranteed to be in **either the** $|+\\rangle$ or **the** $|-\\rangle$ state